tests/data_analyzer/test_beta_2_mode.py

```diff
@@ -340,3 +342,122 @@ def test_DataAnalyzerProcess__formats_and_passes_incoming_stim_packet_through_to
 
     assert outgoing_msg["data_type"] == "stimulation"
     assert outgoing_msg["data_json"] == json.dumps(expected_stim_data)
+
+
+def test_DataAnalyzerProcess__correctly_handles_command_from_main_to_start_mag_analysis(
+    four_board_analyzer_process_beta_2_mode, mocker
+):
+
+    da_process = four_board_analyzer_process_beta_2_mode["da_process"]
+    from_main_queue = four_board_analyzer_process_beta_2_mode["from_main_queue"]
+
+    spied_start_analysis = mocker.spy(da_process, "_start_mag_finding_analysis")
+    mocker.patch.object(data_analyzer, "_mag_finding_analysis_thread", autospec=True)
+
+    test_mag_analysis_command = copy.deepcopy(TEST_START_MAG_ANALYSIS_COMMAND)
+
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_mag_analysis_command, from_main_queue
+    )
+    invoke_process_run_and_check_errors(da_process, 1)
+    spied_start_analysis.assert_called_once_with(TEST_START_MAG_ANALYSIS_COMMAND["content"])
+
+    drain_queue(from_main_queue)
```

Comment:
`invoke_process_run_and_check_errors(da_process, 1)` will grab the command out of `from_main_queue`, so you don't need to drain it here